client/tests/integration/permissions.rs

```diff
@@ -313,3 +313,150 @@ fn permissions_differ_not_only_by_names() {
         .submit_blocking(set_shoes_color)
         .expect("Failed to modify Mouse's shoes");
 }
+
+mod token_parameters {
+    use iroha_data_model::ValueKind;
+
+    use super::*;
+
+    /// Generate tests for permission token granting and for role registration.
+    ///
+    /// `Grant<Account, PermissionToken>` and `Register<Role>` should perform the same checks for
+    /// `PermissionToken` parameters validity. So here we use the same scenarios with different
+    /// runner functions to run tests for both instructions.
+    macro_rules! gen_tests {
```

Comment:
While this is extremely creative, I'd rather we had a sequence of assertions inside one test, rather than multiple. This way you can eliminate the overhead associated to what is essentially a single test scenario.

Reply:
Hm, okay